chapters/interactions-nonlinear.qmd

```diff
@@ -0,0 +1,893 @@
+  label = "ORF",
+  verbose = FALSE
+)
+  
+set.seed(1805)
+pdp_hour <- model_profile(orf_expl, N = 1000, variables = "hour")
+set.seed(1805)
+pdp_day <- model_profile(orf_expl, N = 1000, variables = "day")
+set.seed(1805)
+pdp_int <- model_profile(orf_expl, N = 1000, variables = "hour",
+                         groups = "day")
+
+
+# The PDP data to plot.
+format_mean_prof <- function(x) {
+  dat <-
+    dplyr::as_tibble(x$agr_profiles) %>%
+    dplyr::select(predictor = `_vname_`, value = `_x_`, dplyr::any_of("_groups_"),
+                  prediction = `_yhat_`)
+  # If the 'groups' argument to `model_profile()` is used:
+  if (any(grepl("_groups_", names(dat)))) {
+    dat <- dplyr::rename(dat, group = `_groups_`)
+  }
+  dat
+}
+
+# The individual profiles (with all of the data)
+format_ind_prof <- function(x) {
+  dat <- dplyr::as_tibble(x$cp_profiles)
+  col <- x$agr_profiles$`_vname_`[1]
+  dat$value <- dat[[col]]
+  dat %>%
+    dplyr::select(predictor = `_vname_`, value, prediction = `_yhat_`, id = `_ids_`)
+}                         
+```
+
+```{r}
+#| label: fig-ensemble-pdp
+#| echo: false
+#| warning: false
+#| out-width: 70%
+#| fig-width: 6
+#| fig-height: 6
+#| fig-cap: "Partial dependence profiles for the food delivery hour and day, derived for the tree-based ensemble. Panel (a) shows the individual grid predictions from 1,000 randomly sampled points in the training set. The other panels show the average trends." 
+#| eval: true
+
+day_cols <- c(
+  "#000000FF", "#24FF24FF", "#009292FF", "#B66DFFFF",
+  "#6DB6FFFF", "#920000FF", "#FFB6DBFF"
+)
+
+p_hour_ind <-
+  pdp_hour %>%
+  format_ind_prof() %>%
+  ggplot(aes(value, prediction, group = id)) +
+  geom_line(alpha = 1 / 50) +
+  labs(x = "Order Hour", y = "Delivery Time", title = "(a)")
+
+pred_rng <- range(p_hour_ind$data$prediction)
+
+p_hour <-
+  pdp_hour %>%
+  format_mean_prof() %>%
+  ggplot(aes(value, prediction)) +
+  geom_line(linewidth = 1) +
+  labs(x = "Order Hour", y = NULL, title = "(b)") +
+  ylim(pred_rng)
+
+p_day <-
+  pdp_day %>%
+  format_mean_prof() %>%
+  dplyr::rename(Day = value) %>%
+  ggplot(aes(Day, prediction, col = Day)) +
+  geom_point(cex = 2) +
+  labs(x = "Order Day", y = "Delivery Time", title = "(c)") +
+  ylim(pred_rng) +
+  scale_color_manual(values = day_cols)
+
+p_hour_day <-
+  pdp_int %>%
+  format_mean_prof() %>%
+  dplyr::rename(Day = group) %>%
+  ggplot(aes(value, prediction)) +
+  geom_line(aes(col = Day), linewidth = 1, show_legend = FALSE) +
+  labs(x = "Order Hour", y = NULL, title = "(d)") +
+  ylim(pred_rng) +
+  scale_color_manual(values = day_cols)
+
+(((p_hour_ind + p_hour) / (p_day + p_hour_day)) +
+  plot_layout(guides = "collect")) &
+  theme(legend.position = "bottom")
+```
+
+Friedman’s H-statistic can quantify the effect of interaction terms using partial dependence profiles. When investigating two-factor interactions between predictors $j$ and $j'$, the statistic is 
+
+$$
```

Comment:
There is a lot of abuse of notation here since $f(x_{ij})$ is really a vector. Hopefully, the PDP in the figure helps clarify that.